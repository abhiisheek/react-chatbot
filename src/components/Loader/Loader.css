.loaderIcon {
  width: 6px;
  height: 6px;
  border-radius: 50%;
  position: absolute;
  animation: typing 1s linear infinite alternate;
}

@keyframes typing {
  0% {
    background-color: #9d9d9d;
    box-shadow: 12px 0px 0px 0px #999, 24px 0px 0px 0px #777;
  }
  25% {
    background-color: #999;
    box-shadow: 12px 0px 0px 0px #9d9d9d, 24px 0px 0px 0px #999;
  }
  75% {
    background-color: #777;
    box-shadow: 12px 0px 0px 0px #999, 24px 0px 0px 0px #9d9d9d;
  }
}

.right {
  right: 30px;
}
