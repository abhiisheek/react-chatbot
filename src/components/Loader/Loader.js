import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import styles from './Loader.css';

/** Component for showing loading status */
class Loader extends PureComponent {
  render() {
    const { align, className } = this.props;
    return (
      <div
        className={`${className} ${align === 'right' ? styles.right : 'left'}`}
      />
    );
  }
}

Loader.displayName = 'Loader';

Loader.propTypes = {
  /** Determines which side of the chat window do the loader has to be shown*/
  align: PropTypes.oneOf(['left', 'right']),
  /** CSS Classname for the custom loaders */
  className: PropTypes.string
};

Loader.defaultProps = {
  align: 'left',
  className: `${styles.loaderIcon}`
};

export default Loader;
