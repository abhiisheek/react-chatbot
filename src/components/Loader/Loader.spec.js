import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'enzyme';

import Loader from './Loader';

describe('Loader', () => {
  it('Should have displayName set on the component', () => {
    expect(Loader.displayName).toBeDefined();
  });

  it('Snapshot test', () => {
    const snapshot = renderer.create(<Loader />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('Should render the icon in left by default', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper.instance().props.align).toBe('left');
    expect(wrapper.hasClass('left')).toBeTruthy();
  });

  it('Should render the icon in right when specified right', () => {
    const wrapper = shallow(<Loader align="right" />);
    expect(wrapper.hasClass('right')).toBeTruthy();
  });

  it('Should render the given loader icon when specified', () => {
    const wrapper = shallow(<Loader className="customLoaderIcon" />);
    expect(wrapper.hasClass('customLoaderIcon')).toBeTruthy();
  });
});
