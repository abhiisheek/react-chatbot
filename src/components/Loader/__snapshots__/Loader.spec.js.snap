// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Snapshot test 1`] = `
<div
  className="loaderIcon left"
/>
`;
